src/Relude/Debug.hs

```diff
@@ -51,32 +54,94 @@ import qualified Prelude
 -- trace
 ----------------------------------------------------------------------------
 
--- | Version of 'Debug.Trace.trace' that leaves warning.
+{- | Version of 'Debug.Trace.trace' that leaves warning.
+
+>>> increment l = map (+1) l
+>>> increment [2, 3, 4]
+[3,4,5]
+
+>>> increment l = trace ("incrementing each value of: " ++ show l) (map (+1) l)
+>>> increment [2, 3, 4]
+incrementing each value of: [2,3,4]
+[3,4,5]
+
+-}
 trace :: String -> a -> a
 trace = Debug.trace
 {-# WARNING trace "'trace' remains in code" #-}
 
--- | Version of 'Debug.Trace.traceShow' that leaves warning.
+{- | Version of 'Debug.Trace.traceShow' that leaves warning.
+
+>>> let x = 123
+>>> let f = show
+>>> trace x (f (x + x))
+123
+246
+
+-}
 traceShow :: Show a => a -> b -> b
 traceShow = Debug.traceShow
 {-# WARNING traceShow "'traceShow' remains in code" #-}
 
--- | Version of 'Debug.Trace.traceShowId' that leaves warning.
+{- | Version of 'Debug.Trace.traceShowId' that leaves warning.
+
+>>> traceShowId (1+2+3, "hello" ++ "world")
+(6,"helloworld")
+(6,"helloworld")
+
+-}
 traceShowId :: Show a => a -> a
 traceShowId = Debug.traceShowId
 {-# WARNING traceShowId "'traceShowId' remains in code" #-}
 
--- | Version of 'Debug.Trace.traceShowM' that leaves warning.
-traceShowM :: (Show a, Applicative f) => a -> f ()
-traceShowM = Debug.traceShowM
-{-# WARNING traceShowM "'traceShowM' remains in code" #-}
+{- | Version of 'Debug.Trace.traceM' that leaves warning.
 
--- | Version of 'Debug.Trace.traceM' that leaves warning.
+>>> :{
+  do
+    x <- Just 3
+    traceM ("x: " ++ show x)
+    y <- pure 12
+    traceM ("y: " ++ show y)
+    pure (x*2 + y)
+:}
+
+> x: 3
+> y: 12
```

Comment:
Does this test actually running? :thinking: I don't see character `>` being prepended to tracing calls...

Reply:
no you're right I was using `>` incorrectly!